### PR TITLE
Feature/m UI c 13 send files

### PR DESCRIPTION
**Pull request by eigom**
- File uploading & sending with chat message
- File downloading
- File viewing
- Re-implemented chat message storing as amount of messages is low (and thus memory requirements) and previous storage scheme proved to be a nightmare to extend (i.e. adding new columns to tables etc). **Remove previous app so new DB can be created!**

### CODE REVIEW

**Review by gersonnoboa — COMMENTED** [review]
I think the "Take photo" part is not done yet, right? If that is intended, then ignore.

When uploading an image, I see a square, with no way of removing it, and then when I send the image, the file doesn't get cropped, as I would expect seeing that square.

Also, we only support a certain amount of files. JPEG, JPG, PNG, GIF, DOCX, XLSX, MOV, PDF. I tried adding a ZIP and it uploaded it and then failed. Maybe it could be handled by us, if the files are not of those types, then we fail right away. I will discuss it internally but I don't think anything is needed from your side. Although there is indeed an array somewhere that has other formats we don't support yet.

File: GliaWidgets/Component/ImageView/File/FileImageView.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import UIKit
+#if canImport(QuickLookThumbnailing)
```
[the remark this answers is not in the document]
Comment (eigom) [reply GliaWidgets/Component/ImageView/File/FileImageView.swift:2]: QuickLookThumbnailing is available from iOS 13 but Widgets minimum supported version is 12.

**Comment by eigom** [comment]
> I think the "Take photo" part is not done yet, right? If that is intended, then ignore.
> 
> When uploading an image, I see a square, with no way of removing it, and then when I send the image, the file doesn't get cropped, as I would expect seeing that square.
> 
> Also, we only support a certain amount of files. JPEG, JPG, PNG, GIF, DOCX, XLSX, MOV, PDF. I tried adding a ZIP and it uploaded it and then failed. Maybe it could be handled by us, if the files are not of those types, then we fail right away. I will discuss it internally but I don't think anything is needed from your side. Although there is indeed an array somewhere that has other formats we don't support yet.

Indeed. I kinda missed the Take Photo part. Will check.

Could you please add screenshot of the square/cropped image.

**Comment by gersonnoboa** [comment]
@eigom yes, this is what I am talking about. That square makes me think that the image will be cropped when sending, which is not the case.

image

**Comment by eigom** [comment]
> @eigom yes, this is what I am talking about. That square makes me think that the image will be cropped when sending, which is not the case.
> 
> image

I see. This is due to `UIImagePickerController.allowsEditing = true`. Will set it to `false`.

**Review by VladKupriienko — APPROVED** [review]

File: GliaWidgets/Interactor/Interactor.swift [hunk 2]
```diff
@@ -262,6 +264,11 @@ extension Interactor: Interactable {
         }
     }
 
+    func receive(message: Message) {
+        print("Called: \(#function)")
+        notify(.receivedMessage(message))
+    }
```
[the remark this answers is not in the document]
Comment (eigom) [reply GliaWidgets/Interactor/Interactor.swift:270]: It just seemed sad and lonely in previous location.

File: GliaWidgets/Lib/Upload/FileUploader.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            .map { EngagementFile(id: $0.id) }
+        return Attachment(files: files)
+    }
+    var count: Int { return uploads.count }
+    let state = ObservableValue<State>(with: .idle)
+
+    private var uploads = [FileUpload]()
+    private var storage = FileSystemStorage(directory: .documents)
+
+    init() {}
```
[the remark this answers is not in the document]
Comment (eigom) [reply GliaWidgets/Lib/Upload/FileUploader.swift]: We don't. Removed.

File: GliaWidgets/Lib/Upload/FileUploader.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    func removeAllUploads() {
+        uploads.removeAll()
+        updateState()
+    }
+
+    private func updateState() {
+        var newState: State = .idle
+
+        if uploads.isEmpty {
+            newState = .idle
```
[the remark this answers is not in the document]
Comment (eigom) [reply GliaWidgets/Lib/Upload/FileUploader.swift]: Indeed.

**Comment by gersonnoboa** [comment]
Pff I didn't notice that there are new changes, I am having issues with my email and I have to see everything manually. I apologize. Will check very soon.

**Review by gersonnoboa — APPROVED** [review]
When the app prompts for camera access, it says that it is needed for doing video calls, which is correct. But it never mentions anything about file uploads. Maybe it should be added there, as it could be potentially confusing.

**Comment by gersonnoboa** [comment]
@eigom ah correct. Nothing else from my side. Apologies again for not noticing this before.

**Pull request merged**